libs/ui/src/screens/swap/swap.tsx

```diff
@@ -0,0 +1,240 @@
+  const isApproveSwapButtonDisabled = !isEmpty(errors);
+  const showNavigationBar = !isDefined(fromToken) || !isDefined(toToken) || fromAmount === '';
+  const numberOfRoutes = routes?.flat(1).length ?? 0;
+
+  const isNeedToApprove =
+    isDefined(allowance) &&
+    isDefined(fromToken) &&
+    fromToken.tokenAddress !== GAS_TOKEN_ADDRESS &&
+    isNotEmptyString(fromAmount) &&
+    !isNaN(Number(fromAmount)) &&
+    allowance.lt(parseUnits(fromAmount, fromToken.decimals));
+
+  useEffect(() => {
+    if (isDefined(params) && isDefined(params.fromToken)) {
+      if (params.fromToken.tokenAddress === toToken?.tokenAddress) {
+        return swapTokenOrder();
+      }
+
+      if (isNotEmptyString(fromAmount)) {
+        setValue('fromAmount', getValueWithMaxNumberOfDecimals(fromAmount, params.fromToken.decimals));
+      }
+
+      setValue('fromToken', params.fromToken);
+      clearErrors();
+    }
+
+    if (isDefined(params) && isDefined(params.toToken)) {
+      if (params.toToken.tokenAddress === fromToken?.tokenAddress) {
+        return swapTokenOrder();
+      }
+
+      setValue('toToken', params.toToken);
+    }
+  }, [params]);
+
+  const navigateToSlippageTolerance = () => navigate(ScreensEnum.SlippageTolerance);
+  const navigateToSwapRoute = () =>
+    isDefined(routes) &&
+    isDefined(fromToken) &&
+    isDefined(toToken) &&
+    !loadingRoutes &&
+    navigate(ScreensEnum.SwapRoute, { routes, fromToken, toToken });
+
+  const swapTokenOrder = () => {
+    setValue('fromAmount', toAmount);
+    setValue('toAmount', fromAmount);
+    setValue('fromToken', toToken);
+    setValue('toToken', fromToken);
+
+    clearErrors();
+  };
+
+  const onApproveOrSwapPress = () => {
+    if (loading) {
+      return;
+    }
+
+    if (isNeedToApprove) {
+      return onApprovePress();
+    }
+
+    onSwapPress();
+  };
+
+  return (
+    <ScreenContainer>
+      <HeaderContainer isSelectors>
+        <ScreenTitle title="Swap" onBackButtonPress={goBack} />
+        <View style={styles.header}>
+          {canGetAmountAndRoutes && <Timer getAmountAndRoutes={getAmountAndRoutes} routes={routes} />}
+          <TouchableIcon name={IconNameEnum.Slider} onPress={navigateToSlippageTolerance} />
+        </View>
+      </HeaderContainer>
+
+      <ScreenScrollView>
+        <TokenInput
+          setValue={setValue}
+          trigger={trigger}
+          label="From"
+          control={control}
+          name="fromAmount"
+          token={fromToken}
+          amountInDollar={fromTokenBalance.amountInDollar}
+          tokenParam="fromToken"
+          availableFormattedBalance={fromTokenBalance.availableFormattedBalance}
+          amount={fromAmount}
+          rules={fromTokenRules}
+          availableBalance={fromTokenBalance.availableBalance}
+          error={errors?.fromAmount?.message}
+          errorStyle={errors?.fromAmount?.type === isGreaterThanZeroError ? styles.errorFromAmount : undefined}
```

Comment:
why do we have a custom error style for this case? (this's the only one case you add a long chain of  `errorStyle` prop drilling)

Reply:
we have a long error message for this case and that's why we need extra style for this case